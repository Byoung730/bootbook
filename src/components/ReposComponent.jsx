import React from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';

const ReposComponent = ({repos}) => (
    <div>
        {repos.status === 'FETCHING' ? (
            <div className="alert alert-info">
                <span className="velocity-spinner-small-inline" />
                <span>Getting your repos... </span>
            </div>
        ) : repos.status === 'ERROR' ? (
            <div>
                <div>All bad fam --- error: </div>
                <div>
                    <strong>{repos.error.message}</strong>
                </div>
            </div>
        ) : repos.status === 'SUCCESS' ? (
            <ul className="list-group">
                {repos.value.map(repo => (
                    <a
                        key={repo.id}
                        href={repo.html_url}
                        className="list-group-item"
                    >
                        {repo.name}
                    </a>
                ))}
            </ul>
        ) : null}
    </div>
);

ReposComponent.displayName = 'ReposComponent';

ReposComponent.propTypes = {
    repos: PropTypes.instanceOf(Immutable.Record).isRequired
};

export default ReposComponent;
