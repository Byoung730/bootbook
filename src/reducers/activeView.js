// const action = {
//     type: 'SET_ACTIVE_VIEW',
//     view: 'MY_STATUSES'
// };

export default (state = 'MY_STATUSES', action) => {
    switch (action.type) {
        case 'SET_ACTIVE_VIEW':
            return action.view;
        default:
            return state;
    }
};
