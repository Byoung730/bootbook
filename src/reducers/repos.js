import Immutable from 'immutable';

export const ReposRecord = Immutable.Record({
    status: null,
    error: null,
    value: null
});

export default (state = new ReposRecord(), action) => {
    switch (action.type) {
        case 'REPOS_FETCHING':
            return state.set('status', 'FETCHING');

        case 'REPOS_SUCCESS':
            return new ReposRecord({
                status: 'SUCCESS',
                error: null,
                value: action.response
            });

        case 'REPOS_ERROR':
            return new ReposRecord({
                status: 'ERROR',
                error: action.error
            });

        default:
            return state;
    }
};
