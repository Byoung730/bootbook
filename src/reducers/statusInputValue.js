export default (state = '', action) => {
    switch (action.type) {
        case 'POST_NEW_STATUS':
            return action.status;
        default:
            return state;
    }
};
