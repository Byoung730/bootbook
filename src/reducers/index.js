import {combineReducers} from 'redux';
import activeView from './activeView';
import statusInputValue from './statusInputValue';
import statuses from './statuses';
import repos from './repos';

export default combineReducers({
    activeView,
    statuses,
    statusInputValue,
    repos
});
