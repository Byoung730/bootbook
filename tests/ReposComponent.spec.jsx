import ReposComponent from '../src/components/ReposComponent';
import {ReposRecord} from '../src/reducers/repos';
import {shallow} from 'enzyme';
import React from 'react';

jest.unmock('../src/components/ReposComponent');
jest.unmock('../src/reducers/repos');

describe('ReposComponent', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<ReposComponent repos={new ReposRecord()} />);
    });

    it('Default view is empty', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Displays alert with spinner when repos are fetching', () => {
        const newReposRecord = new ReposRecord({
            status: 'FETCHING'
        });
        wrapper.setProps({repos: new ReposRecord()});

        expect(wrapper).toMatchSnapshot();
    });
});
