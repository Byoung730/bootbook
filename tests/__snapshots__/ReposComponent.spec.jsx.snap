// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposComponent Default view is empty 1`] = `<div />`;

exports[`ReposComponent Displays alert with spinner when repos are fetching 1`] = `<div />`;
